### PR TITLE
namespace: add name spec in cephBlockPoolRadosNamespace CRD

### PR DESCRIPTION
**Pull request by parth-gr**
instead of use metadata name as the backend resource name we need to add a new name in the spec which can be the actual backend name

Closes: https://github.com/rook/rook/issues/13220

**Which issue is resolved by this Pull Request:**
Resolves #  https://github.com/rook/rook/issues/13220

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/radosnamespace.yaml [hunk 1]
```diff
@@ -5,5 +5,7 @@ metadata:
   name: namespace-a
   namespace: rook-ceph # namespace:cluster
 spec:
+  # for multiple blockpool if same radosnamespace name is required use radosNamespaceName
+  # radosNamespaceName: namespace-a
```
[the remark this answers is not in the document]
Comment (parth-gr) [reply deploy/examples/radosnamespace.yaml]: Updated

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 2]
```diff
@@ -2919,6 +2919,9 @@ type CephBlockPoolRadosNamespaceList struct {
 
 // CephBlockPoolRadosNamespaceSpec represents the specification of a CephBlockPool Rados Namespace
 type CephBlockPoolRadosNamespaceSpec struct {
+	// The name of the CephBlockPoolRadosNamespaceSpec namespace. If not set, the default is the name of the CR.
+	// +optional
+	Name string `json:"radosNamespaceName,omitempty"`
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/types.go:2924]: We need the `name` to match in the yaml
```suggestion
	Name string `json:"name,omitempty"`
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/apis/ceph.rook.io/v1/types.go (hunk 2 above)
Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go]: Updated

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 3]
```diff
@@ -2919,6 +2919,9 @@ type CephBlockPoolRadosNamespaceList struct {
 
 // CephBlockPoolRadosNamespaceSpec represents the specification of a CephBlockPool Rados Namespace
 type CephBlockPoolRadosNamespaceSpec struct {
+	// The name of the CephBlockPoolRadosNamespaceSpec namespace. If not set, the default is the name of the CR.
+	// +optional
+	Name string `json:"name,omitempty"`
```
Comment (Rakshith-R) [inline pkg/apis/ceph.rook.io/v1/types.go:2924]: Let's add CEL rules to make this immutable. 
https://github.com/csi-addons/kubernetes-csi-addons/blob/main/apis/replication.storage/v1alpha1/volumereplication_types.go#L71

make blockPoolName below too (this can be another pr).

Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: ok

Comment (travisn) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: Note that #13177 needs to merge before we can merge other CEL rules.

Comment (travisn) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: #13177 merged, want to add the CEL rule now?

Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: Will open a separate PR for both svgs and radosnamesapce for adding this

Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: And it also need to add cel rules in the pool, which wont be right way to add them

Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: Added a tracker https://github.com/rook/rook/issues/13315

Comment (parth-gr) [reply pkg/apis/ceph.rook.io/v1/types.go:2965]: @travisn

**Review by travisn — APPROVED** [review]
